src/app/pages/LendingPage/components/LendingDialog/index.tsx

```diff
@@ -125,26 +125,23 @@ export function LendingDialog({
     setAmount('');
   }, [currency]);
 
-  const disabled = () => (type === 'add' ? !valid : !validRedeem);
+  const disabled = () => (type === 'add' ? !isValid : !isValidRedeem);
 
   const errorMessage = useMemo(() => {
     if (type === 'add') {
-      if (!greaterZero) return t(translations.validationErrors.minimumZero);
-      if (!enoughBalance)
+      if (!isGreaterThanZero)
+        return t(translations.validationErrors.minimumZero);
+      if (!hasSufficientBalance)
         return t(translations.validationErrors.insufficientBalance);
     }
-  }, [type, greaterZero, t, enoughBalance]);
+  }, [type, isGreaterThanZero, t, hasSufficientBalance]);
 
   const contractName = getLendingContractName(currency);
   const tokenAddress = getTokenContract(currency).address;
-  const methodName =
-    type === 'add'
-      ? currency === Asset.RBTC
-        ? 'mintWithBTC'
-        : 'mint'
-      : currency === Asset.RBTC
-      ? 'burnToBTC'
-      : 'burn';
+  const getMethodName = () => {
```

Comment:
Use `useCallback` here and please, don't write conditions and return on one line (if it's not a ternary expression).

Just add `{}` and some spacing.